### PR TITLE
RUM-7180 SwiftUI Benchmark Scenario

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Add a Session Replay SwiftUI scenario for continuous benchmarking.

**Note:** Everything under `BenchmarkTests/SwiftUICatalog` is 3rd party.

### How?

Integrate with the third-party [SwiftUICatalog](https://github.com/barbaramartina/swiftuicatalog) in a similar fashion than [UIKitCatalog](https://developer.apple.com/documentation/uikit/mac_catalyst/uikit_catalog_creating_and_customizing_views_and_controls).

The scenario will go through multiple SwiftUI components to measure pre-defined metrics.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes
- [ ] Add Objective-C interface for public APIs (see our [guidelines](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3157787243/RFC+-+Modular+Objective-C+Interface#Recommended-solution) [internal]) and run `make api-surface`)

### CODE REVIEW

**Comment by ncreated** [comment]
@maxep Would it be possible to consider bundling the images differently, skipping some, or resizing them to help reduce the impact on the repository size? The total size of the images in this PR is 1.5MB, which adds over 10% to the current size of the repository (<15MB).

**Comment by maxep** [comment]
@ncreated Yes, totally 👍

**Review by simaoseica-dd — APPROVED** [review]

File: BenchmarkTests/SwiftUICatalog/ContentView.swift [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    var popovers: some View {
+        Section(header: Text("Popovers, alerts and sheets")
+            .font(.title)
+            .modifier(ListSectionFontModifier())) {
+                
+//                Link(destination: PopoversComponentView(),
+//                     label: "Popovers")
+                Link(destination: SheetsView().modifier(monitor.viewModifier(name: "Sheets")),
+                     label: "Sheets")
+//                Link(destination: AlertsComponentView(),
```
Comment (simaoseica-dd) [inline BenchmarkTests/SwiftUICatalog/ContentView.swift:307]: question/ Do you want the links hidden? A configuration with a toggle might help here

File: BenchmarkTests/BenchmarkTests.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1950,7 +1947,7 @@
 				INFOPLIST_KEY_UILaunchStoryboardName = LaunchScreen;
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPad = "UIInterfaceOrientationPortrait UIInterfaceOrientationPortraitUpsideDown UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight";
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPhone = "UIInterfaceOrientationPortrait UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight";
-				IPHONEOS_DEPLOYMENT_TARGET = 13.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 17.0;
```
[the remark this answers is not in the document]
Comment (maxep) [reply BenchmarkTests/BenchmarkTests.xcodeproj/project.pbxproj:1943]: We can only define the minimum target. Our Synthetics Tests runs on iOS 17 devices so we don't loose coverage by bumping this version.

File: BenchmarkTests/SwiftUICatalog/ContentView.swift (hunk 1 above)
Comment (maxep) [reply BenchmarkTests/SwiftUICatalog/ContentView.swift:303]: They are not actually implemented, we can keep them hidden and add them later 👍

**Review by mariedm — APPROVED** [review]

File: BenchmarkTests/Runner/Scenarios/SessionReplay/SessionReplaySwiftUIScenario.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                applicationID: info.applicationID
+            )
+        )
+
+        SessionReplay.enable(
+            with: SessionReplay.Configuration(
+                replaySampleRate: 100,
+                textAndInputPrivacyLevel: .maskSensitiveInputs,
+                imagePrivacyLevel: .maskNone,
+                touchPrivacyLevel: .show
```
[the remark this answers is not in the document]
Comment (maxep) [reply BenchmarkTests/Runner/Scenarios/SessionReplay/SessionReplaySwiftUIScenario.swift:39]: Ah, good catch!!

File: BenchmarkTests/BenchmarkTests.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1197,7 +1969,7 @@
...
```
Comment (ncreated) [inline BenchmarkTests/BenchmarkTests.xcodeproj/project.pbxproj:1972]: suggestion/ We have two `xcconfig` files set up for Benchmarks project, this setting could easily go there to simplify configuration.

**Review by ncreated — APPROVED** [review]
Looks good 👌. My only ask is to update the `LICENSE-3rdparty.csv`. I think it should also mention the `UIKitCatalog`.

File: BenchmarkTests/SwiftUICatalog/LICENSE
[the remark this answers is not in the document]
Comment (maxep) [reply BenchmarkTests/SwiftUICatalog/LICENSE:1]: Good call, done!

File: BenchmarkTests/BenchmarkTests.xcodeproj/project.pbxproj (hunk 4 above)
Comment (maxep) [reply BenchmarkTests/BenchmarkTests.xcodeproj/project.pbxproj]: The `xcconfig` is currently applied to the `Runner` target only.
What I did instead is set the `IPHONEOS_DEPLOYMENT_TARGET` at the project level and remove overrides in the targets.

**Review by mariedm — APPROVED** [review]

**Pull request merged**